### DISCUSSION TITLE
Use SmartCacheDataset to cache on GPU?

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/tutorials#1407] [Q&A]
Hi!
Can we use `SmartCacheDataset` to cache intermediate data (i.e., after going through a chain of deterministic transforms) on GPU memory? I saw `CacheDataset` is used in this [fast_training_tutorial](https://github.com/Project-MONAI/tutorials/blob/main/acceleration/fast_training_tutorial.ipynb), so wonder if `SmartCacheDataset` can be used as well.

**Comment by KumoLiu** [comment]
Hi @hw-ju, yes `SmartCacheDataset` can be used as well. Just make sure you have used it correctly.
https://github.com/Project-MONAI/MONAI/blob/84940298bff9ff1f54845eb8ae7422c0092c09c1/monai/data/dataset.py#L957-L975
Thanks!

**Reply by hw-ju** [reply]
@KumoLiu Thanks a lot :D!
